src/operations/create_collection.ts

```diff
@@ -100,6 +103,74 @@ export class CreateCollectionOperation extends CommandOperation<Collection> {
     const name = this.name;
     const options = this.options;
 
+    const encryptedFields =
+      options.encryptedFields ??
+      db.s.client.options.autoEncryption?.encryptedFieldsMap?.[`${db.databaseName}.${name}`];
+    if (!encryptedFields) {
+      return this.executeWithoutEncryptedFieldCheck(server, session, callback);
+    }
+
+    const escCollection = encryptedFields.escCollection || `enxcol_.${name}.esc`;
+    const eccCollection = encryptedFields.eccCollection || `enxcol_.${name}.ecc`;
+    const ecocCollection = encryptedFields.ecocCollection || `enxcol_.${name}.ecoc`;
+    new CreateCollectionOperation(db, escCollection).executeWithoutEncryptedFieldCheck(
+      server,
+      session,
+      err => {
+        if (err) {
+          return callback(err);
+        }
+
+        new CreateCollectionOperation(db, eccCollection).executeWithoutEncryptedFieldCheck(
+          server,
+          session,
+          err => {
+            if (err) {
+              return callback(err);
+            }
+
+            new CreateCollectionOperation(db, ecocCollection).executeWithoutEncryptedFieldCheck(
+              server,
+              session,
+              err => {
+                if (err) {
+                  return callback(err);
+                }
+
+                this.executeWithoutEncryptedFieldCheck(server, session, (err, coll) => {
+                  if (err) {
+                    return callback(err);
+                  }
+
+                  new CreateIndexOperation(db, name, { __safeContent__: 1 }, {}).execute(
+                    server,
+                    session,
+                    err => {
+                      if (err) {
+                        return callback(err);
+                      }
+
+                      return callback(undefined, coll);
+                    }
+                  );
+                });
+              }
+            );
+          }
+        );
+      }
+    );
```

Comment:
Is it okay/preferred to perform these operations as part of another `CommandOperation` object, or would it make more sense to move this into the `db.createCollection()` wrapper that calls `executeOperation()`?

The latter a bit unappealing to me because it would mean taking care of the promise/callback handling manually, but as far as I understand, there are concrete differences between the two approaches in that `executeOperation()` performs server selection, so there is probably a “correct” choice. Please tell me if I made the wrong one :slightly_smiling_face:

Reply:
Per our discussion we thought it better to do this in the operation object so server selection would ensure that all the operations execute on the same server instead of server selection getting run for each individual command.